src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -106,12 +110,31 @@ public int Count
             }
         }
 
-        public Enumerator GetEnumerator()
+        public T this[int index]
```

Comment:
Have we already discussed the potential for this returning `ref T` instead of `T`? e.g.
```C#
public ref T this[int index]
{
    get
    {
        if ((uint)index >= (uint)_count) ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.index);
        return ref _array[_offset + index];
    }
}
```
?